_studio/mfx_lib/scheduler/include/mfx_scheduler_core.h

```diff
@@ -312,6 +313,31 @@ class mfxSchedulerCore : public MFXIScheduler2
     inline MFX_SCHEDULER_THREAD_CONTEXT* GetThreadCtx(mfxU32 thread_id)
     { return &m_pThreadCtx[thread_id]; }
 
+    // Invokes functor 'bool F(MFX_SCHEDULER_TASK*)' for every valid task that returns 'false' to continue iteration or 'true' to stop it.
+    template <typename F>
+    void ForEachTask(F&& f)
```

Comment:
I like the idea, but I don't like template name since I expect from it to iterate for really all the tasks in the queue while you have boolean functor which signals the stop. Can you think about rename? I am honestly not sure in how I would name it. Also, I would suggest to have 2 templates one is the same you have here and another which is real for_each:
```
template <typename F> // bool F();
void ForEachTaskWhile(F&& f) {}  // that's the one you have in PR

template <typename F> // void F();
void ForEachTask(F&& f) {
    return ForEachTaskWhile([] { F(); return false;} );
}
```

If you will follow "*While" name, consider to revert boolean meaning for `true` to continue to iterate like in `do {} while(true)` syntax, i.e. "For Each Task While Predicate returns True".